src/harness/reference_models/aggregate_interference/aggregate_interference.py

```diff
@@ -0,0 +1,362 @@
+    {(latitude, longitude, low_frequency, high_frequency,entity_type):
+    aggregate_interference(dBm)}
+  """
+
+  grant_list = interf.getAllGrantInformationFromCbsdDataDump(cbsd_list)
+
+  # Get the protection point of the ESC
+  esc_point = esc_record['installationParam']
+  protection_point = (esc_point['latitude'], esc_point['longitude'])
+
+  # Get azimuth radiation pattern
+  ant_pattern = esc_point['azimuthRadiationPattern']
+
+  # Initialize array of size length of ant_pattern list
+  ant_pattern_gain = np.arange(len(ant_pattern))
+  for i, pattern in enumerate(ant_pattern):
+    ant_pattern_gain[i] = pattern['gain']
+
+  # Get ESC antenna information
+  esc_antenna_info = interf.EscInformation(antenna_height=esc_point['height'],
+                       antenna_azimuth=esc_point['antennaAzimuth'],
+                       antenna_gain=esc_point['antennaGain'],
+                       antenna_pattern_gain=ant_pattern_gain)
+
+  # ESC Category A CBSDs between 3550 - 3660 MHz within 40 km
+  protection_channels = interf.getProtectedChannels(interf.ESC_CAT_A_LOW_FREQ_HZ, 
+                          interf.ESC_CAT_A_HIGH_FREQ_HZ)
+  aggregate_interference = aggregateInterferenceForPoint(protection_point, 
+                             protection_channels, interf.ESC_CAT_A_LOW_FREQ_HZ, 
+                             interf.ESC_CAT_A_HIGH_FREQ_HZ, grant_list, None, 
+                             esc_antenna_info, interf.ProtectedEntityType.ESC_CAT_A)
+  # ESC Category B CBSDs between 3550 - 3680 MHz within 80 km
+  protection_channels = interf.getProtectedChannels(interf.ESC_CAT_B_LOW_FREQ_HZ, 
+                          interf.ESC_CAT_B_HIGH_FREQ_HZ)
+  aggregate_interference = aggregateInterferenceForPoint(protection_point, 
+                             protection_channels, interf.ESC_CAT_B_LOW_FREQ_HZ, 
+                             interf.ESC_CAT_B_HIGH_FREQ_HZ, grant_list, None, 
+                             esc_antenna_info, interf.ProtectedEntityType.ESC_CAT_B)
+
+  return aggregate_interference
+
+
+def calculateAggregateInterferenceForGwpz(gwpz_record, cbsd_list):
+  """Calculates aggregate interference for GWPZ.
+  
+  Calculates aggregate interference for each protection constraint in the GWPZ
+  based on the interferences caused by CBSD's grants in the neighborhood of
+  the GWPZ protection entity.
+
+  Args:
+    gwpz_record: GWPZ protection entity
+    cbsd_list: list of CBSD objects containing RegistrationRequests and GrantRequests
+  Returns:
+    Aggregate interference to GWPZ in the dictionary format,where the key is a tuple. 
+    {(latitude, longitude, low_frequency, high_frequency,entity_type):
+    aggregate_interference(dBm)}
+  """
+
+  grant_list = interf.getAllGrantInformationFromCbsdDataDump(cbsd_list)
+  gwpz_region = gwpz_record['zone']['features'][0]['properties']['gwpzRegionType']
+
+  # Get Fine Grid Points for a GWPZ protection area
+  protection_points = utils.GridPolygon(gwpz_record)
```

Comment:
This function requires the resolution in arcsec. So you should pass 2.

Reply:
Updated   resolution in arcsecs to 2